core/src/main/scala/spinal/core/internals/ComponentEmitterVerilog.scala

```diff
@@ -54,6 +54,7 @@ class ComponentEmitterVerilog(
   val definitionAttributes  = new StringBuilder()
   val beginModule = new StringBuilder()
   val declarations = new StringBuilder()
+  var declaredInterface: Set[SVIF] = Set()
```

Comment:
I would say to avoid imutable Set for performance reasons, and instead go for either mutable.HashSet if you know you will never iterate on the set.
But if you iterate on the set it is very important to use mutable.LinkedHashSet in order to preserve the insertion order, allowing to remain deterministic.